### PR TITLE
merge for 1.0

### PR DESCRIPTION
**Pull request by jcupitt**
possible 1.0.0 ready for review

Changes:

* add PSR-3 logging system
* switch to PSR-2 formatting
* removed `Enum`
* added `Config` and `Utils`
* more tests
* another example

### CODE REVIEW

**Comment by kleisauke** [comment]
I've added few inline comments. Is the version 1.0.0 still marked as experimental? If not then we can delete the `EXPERIMENTAL` file in the dev branch.

Overall, I think this is ready for releasing. :shipit: 
(Features such as continuous integration (Travis CI), Windows support and `gh-pages` docs can be added in the next release, or in any further release).

File: src/Image.php [hunk 1]
```diff
@@ -1509,6 +1509,7 @@ public function bandand(): Image
      */
     public function bandor(): Image
     {
+        // phpdoc hates OperationBoolean::OR, so use the string form here
```
Comment (jcupitt) [inline src/Image.php]: I don't think so, phpcs says that `OperationBoolean::OR` is a syntax error and stops parsing.

**Comment by jcupitt** [comment]
OK, all comments addressed I think. I revised the README too.

I'll try to fix the memory problem in `php-vips-ext` before doing any publicity, and we should probably have some formatted docs up somewhere too.

**Comment by jcupitt** [comment]
Also, thank you both very much! Hooray!

**Pull request merged**